src/main/java/com/parsing/pdf/parsing/ParsingPDFService.java

```diff
@@ -0,0 +1,142 @@
+    private static final int WHITE_PIXEL_CODE = 255;
+    private final RectangleDetector rectangleDetector;
+    private final DataRecognizer dataRecognizer;
+    private final TableDetector tableDetector;
+    private final String DIR_TO_READ_TESSDATA = "/tessdata/";
+    private List<Row> table = new ArrayList<>();
+
+    public String parseProzorroFile(MultipartFile file) throws IOException, TesseractException {
+        OpenCV.loadLocally();
+
+        PDDocument document = PDDocument.load(file.getBytes());
+        String lastPagePDF = getLastPagePDF(document);
+        String fileTableName = tableDetector.detectTable(lastPagePDF);
+        table = rectangleDetector.detectRectangles(fileTableName);
+        table = extractTextFromScannedDocument(fileTableName);
+        dataRecognizer.recognizeLotPDFResult(table);
+        JSONObject obj = new JSONObject();
+        obj.put("fileName", file.getOriginalFilename());
+        StringBuilder builder = new StringBuilder();
+        for (Row row : table) {
+            for (Column column : row.getColumns()) {
+                builder.append(column.getParsingResult());
+            }
+        }
+        obj.put("text", builder);
+        return obj.toString();
+    }
+
+    private List<Row> findStrippedTextFromFile(MultipartFile file) {
+        try {
+            PDDocument document = PDDocument.load(file.getBytes());
+            PDFTextStripper stripper = new PDFTextStripper();
+            String strippedText = stripper.getText(document);
+            if (strippedText.trim().isEmpty()) {
+                return Collections.emptyList();
+            }
+        } catch (Exception e) {
+            throw new PDFParsingException(String.format("Parsing for file: %s failed.", file.getOriginalFilename()));
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Row> extractTextFromScannedDocument(String fileTableName) throws TesseractException {
+        ITesseract itesseract = new Tesseract();
+        itesseract.setDatapath(getPathTessData());
+        itesseract.setLanguage("ukr+eng");
+        final Mat tableMat = Imgcodecs.imread(fileTableName);
+        for (Row row : table) {
+            for (Column column : row.getColumns()) {
+                Mat mat = fillSquareWhitePixel(tableMat, column.getRectangle());
+                Imgcodecs.imwrite(table.indexOf(row) + " " + row.getColumns().indexOf(column) + ".png", mat);
+            }
+        }
+        for (Row row : table) {
+            for (Column column : row.getColumns()) {
+                String filename = table.indexOf(row) + " " + row.getColumns().indexOf(column) + ".png";
+                String result = itesseract.doOCR(new File(filename));
+                column.setParsingResult(result);
+                log.info(filename + " = " + result);
+            }
+        }
+        return table;
+    }
+
+    private Mat fillSquareWhitePixel(Mat image, Rectangle rectangle) {
```

Comment:
not really understand what this method does by it's name.

Reply:
changed to cleanTableBorders.